subsys/storage/nvcb/nvcb.c

```diff
@@ -0,0 +1,882 @@
+/*
+ * Non volatile circular buffer
+ *
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/storage/nvcb.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(fs_nvcb, CONFIG_NVCB_LOG_LEVEL);
+
+#define NVCB_ENTRYHDRS        0b01011000 /* cbor byte string, 1 byte length */
```

Comment:
Does this mean that the entry length is always 24 bytes for this type of entry
According to https://datatracker.ietf.org/doc/html/rfc8949 0b010_xxxxx represents the Major type 2, where xxxxx is the length if it is <= 24

Reply:
For length < 24 there are multiple Major type 2 that should be accepted. For bytestrings with length < 24 the length can be added to the header (most efficient), but also a 2 (or even a 4) byte  length specifier are valid representations. To fully support the off chip generation (that would generate the most efficient data size) it is better to add specific handling of these smaller items. I will add this as when it is clear that the proposed nvcb will not be rejected.